subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java

```diff
@@ -246,7 +246,7 @@ public void afterStart(GradleInternal gradle) {
                 @Override
                 public void beforeComplete(GradleInternal gradle) {
                     if (isRetentionEnabled(gradle.getStartParameter().getSystemPropertiesArgs())) {
-                        virtualFileSystem.startWatching(Collections.singleton(gradle.getRootProject().getProjectDir().toPath()));
+                        virtualFileSystem.startWatching(Collections.singleton(gradle.getRootProject().getProjectDir().getAbsolutePath()));
```

Comment:
I'd love to keep using a strong type to pass around file system locations. Ideally I'd use `Path`, but `File` is close second. Using `String` is something I'd like to avoid if possible.

Reply:
I am using `File` now for `mustWatchDirectories`. I agree that it would be nice to use a stronger type than `String` everywhere, though creating a `File` or a `Path` from a `String` is quite expensive, so we can't really use them on the lower level APIs.